### DISCUSSION TITLE
How to trigger a realtime event, when using pocketbase as a framework?

### DISCUSSION THREAD

**Discussion by MarvinJWendt** [pocketbase/pocketbase#802] [Q&A] [answered]
Hi, I am currently using pocketbase as a framework. I have a script that inputs some mock data into my database:

```go
		fake := faker.NewFaker()
		productsCollection, err := app.Dao().FindCollectionByNameOrId("products")
		if err != nil {
			return err
		}

		product := models.NewRecord(productsCollection)
		err = product.Load(map[string]any{
			"name":        fake.RandomProduct(),
			"description": fake.RandomLoremParagraph(),
			"price":       fake.RandomIntBetween(1_00, 100_00),
			"displayname": fake.RandomProduct(),
		})
		if err != nil {
			return err
		}

		err = app.Dao().SaveRecord(product)
		if err != nil {
			return err
		}
```

It seems, that this does not trigger the realtime updates, as my website does not receive any updates. It works when I change records on the admin UI. Thanks!

PS: This is my frontend code to verify the subscription: 

```js
  client.realtime.subscribe('products', (data) => {
    console.log(data)
    refreshProducts()
  })
```

**Comment by ganigeorgiev** [comment, marked as answer]
What version of PocketBase are you using?

You example code should work since the realtime api is attached to the `OnModelAfterCreate` and `OnModelAfterUpdate` event hooks - https://github.com/pocketbase/pocketbase/blob/master/apis/realtime.go#L216-L235.

You can also trigger the hooks manually by calling something like:

```go
app.OnModelAfterCreate().Trigger(&core.ModelEvent{
    Dao: app.Dao(),
    Model: product,
})
```

**Reply by MarvinJWendt** [reply]
Oh, I think this is because the mock data script is running as its own instance that connects to the same SQLite file. I injected a "mock" command into Cobra and while one process is "serving" the mock command modifies the DB. I somehow thought pocketbase would listen to the DB changes. Is there any way to externally trigger a broadcast? I don't think the API can do that, yet. (?)

**Reply by ganigeorgiev** [reply]
In PocketBase all hooks are triggered by application level code. We are not using db triggers because they are very limited and it is hard to control when not to fire inside a transaction (also often there is a need for additional context data that you can't get from the database record).

> Is there any way to externally trigger a broadcast?

The easiest way would be to send a `POST /api/collections/products/records` request using curl or if you want to execute it inside go - using the std go http client (you may have to authorize as admin beforehand, depending whether your collection has restricted create).

**Reply by MarvinJWendt** [reply]
Okay, got it! Thank you.
